pkg/linter/linter.go

```diff
@@ -67,9 +67,27 @@ type Linter struct {
 	profiling            bool
 	instrumentation      bool
 	hasCustomRules       bool
+	isPrepared           bool
```

Comment:
So a `Linter` here is not supposed to be used by multiple callers concurrently? Only a prepared Linter is?

If so, it could make sense to introduce another type for that, so we have
```go
func (l *Linter) Prepare(context.Context) (*PreparedLinter, error)
```

Reply:
You're one step ahead of me, as concurrent use isn't really something I've considered yet. All instances of Linter should be safe for concurrent use I think, as we're not using a pointer receiver, and all "writes" to the linter return a new instance. The difference would be that `Prepare` returns a new `Linter` while `Lint` returns a report.

`isPrepared` is mostly there to have something to track the state of the linter, and to ensure that while the same prepared linter can be reused for multiple inputs, changing something like configuration options should trigger a "reprepare". Does that make sense?

Perhaps making `Linter` and interface and have two different implementations (like the PreparedLinter you suggested) would be more elegant 🤔